src/Transaction.js

```diff
@@ -3,12 +3,38 @@
  * An account of what has happened.
  */
 
-export default function Transaction (type, payload, complete) {
-  return {
-    type     : `${ type }`,
-    error    : false,
-    active   : arguments.length > 1,
-    payload  : payload,
-    complete : complete
+import coroutine  from './coroutine'
+import flatten    from './flatten'
+import eventually from './eventually'
+
+export default function Transaction (action, payload) {
+  this.action  = action
+  this.type    = `${ action }`
+  this.active  = arguments.length > 1
```

Comment:
The payload might be `undefined`, but we only use the second argument
internally to Microcosm. I'll give this a second look.

On Tue, Jan 26, 2016 at 11:23 AM, Lawson Kurtz <notifications@github.com>
wrote:

> In src/Transaction.js
> <https://github.com/vigetlabs/microcosm/pull/94#discussion_r50857981>:
>
> > @@ -3,12 +3,38 @@
> >   * An account of what has happened.
> >   */
> >
> > -export default function Transaction (type, payload, complete) {
> > -  return {
> > -    type     : `${ type }`,
> > -    error    : false,
> > -    active   : arguments.length > 1,
> > -    payload  : payload,
> > -    complete : complete
> > +import coroutine  from './coroutine'
> > +import flatten    from './flatten'
> > +import eventually from './eventually'
> > +
> > +export default function Transaction (action, payload) {
> > +  this.action  = action
> > +  this.type    = `${ action }`
> > +  this.active  = arguments.length > 1
>
> just curious: why not check for the definedness of payload instead?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/vigetlabs/microcosm/pull/94/files#r50857981>.
>